CRUD-API/lib/DataTypes/product/index.js

```diff
@@ -28,40 +28,58 @@ class Product {
       productName,
       productDescription
    ) {
-      if (
-         getType(productNumber) == types.Number &&
-         getType(cost) == types.Number &&
-         getType(markup) == types.Number &&
-         getType(sellAtList) == types.Boolean &&
-         getType(productName) == types.String &&
-         getType(productDescription) == types.String
-      ) {
-         this.number = productNumber;
-         this.cost = cost;
-         this.markup = markup >= this.MIN_MARKUP ? markup : this.MIN_MARKUP;
-         this.listPrice = listPrice;
-         this.sellAtList = sellAtList;
-         this.productName = productName;
-         this.productDescription = productDescription;
-      } else {
-         console.error(
-            'Invalid arguments.',
-            productNumber,
-            cost,
-            markup,
-            listPrice,
-            sellAtList,
-            productName,
-            productDescription
-         );
+      switch (true) {
+         case getType(productNumber) == types.Number &&
+            getType(cost) == types.Number &&
+            getType(markup) == types.Number &&
+            getType(sellAtList) == types.Boolean &&
+            getType(productName) == types.String &&
+            getType(productDescription) == types.String:
+            this.number = productNumber;
+            this.cost = cost;
+            this.markup = markup >= this.MIN_MARKUP ? markup : this.MIN_MARKUP;
+            this.listPrice = listPrice;
+            this.sellAtList = sellAtList;
+            this.productName = productName;
+            this.productDescription = productDescription;
+
+            // Add a flag for override
+            this.overrideCheck = false;
+
+            // Check if price is greater than cost
+            if (!this.overrideCheck && this.price <= this.cost) {
```

Comment:
Was this tested? Accessing a getter in construction can cause some wonky behavior.

Reply:
Yes , because the getter may rely on other properties that have not been fully initialized yet. i have fixed this

Reply:
I see; however, it's generally not a good practice to access getters/setters during construction, as the structure has not been fully constructed.

Reply:
I addressed a critical aspect of my functionality by determining the appropriate price for a product. I achieve this by directly computing the price, factoring in elements like cost, markup percentage, and the choice to sell at a listed price. If the product is designated to be sold at the listed price, I utilize this value; otherwise, I calculate the price by multiplying the cost by the markup percentage. To maintain profitability, I verify that the calculated price surpasses the cost; if not, I issue an error message and halt further execution. Overall, this code segment equips me with a robust mechanism for precisely pricing products, ensuring a reasonable profit margin and contributing to effective product management. 

This is my new solution , what do you think about this

Reply:
Based on my new changes the test passed , earlier i didn't test for first implementation
![Screenshot 2023-10-11 at 9 39 06 PM](https://github.com/bj-rutledge/CRUD-API-Example/assets/118662574/d11e57f0-92f6-403c-a038-050b2dadbf7a)